File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/HomeworkActAdm/adapterRecVHomeworkActAdm.java
package com.example.adminapp.HomeworkActAdm;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.adminapp.R;

import java.util.ArrayList;

public class adapterRecVHomeworkActAdm extends RecyclerView.Adapter<adapterRecVHomeworkActAdm.myviewholder> {

    ArrayList<dataModelRecVHomeworkActAdm> dataHolder;
    ArrayList<dataModelRecVHomeworkActAdm> dataHolderBackup;

    public adapterRecVHomeworkActAdm(ArrayList<dataModelRecVHomeworkActAdm> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view  = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.single_row_des_homeworkdefault_fragadm,parent,false);
        return new myviewholder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull myviewholder holder, int position) {
        holder.hwTitleHomeworkDefaultFragAdm.setText(dataHolder.get(position).getHwTitleHomeworkDefaultFragAdm());
    }

    @Override
    public int getItemCount() {
        return dataHolder.size();
    }

    public class myviewholder extends RecyclerView.ViewHolder {
        TextView hwTitleHomeworkDefaultFragAdm;
        public myviewholder(@NonNull View itemView) {
            super(itemView);
            hwTitleHomeworkDefaultFragAdm = itemView.findViewById(R.id.homeTitleTvSingleRdHomeworkDefaultFragAdm);
        }
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/HomeworkActAdm/adapterRecVHomeworkActAdm.kt
package com.example.adminapp.HomeworkActAdm

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R

class adapterRecVHomeworkActAdm(var dataHolder: ArrayList<dataModelRecVHomeworkActAdm>) :
    RecyclerView.Adapter<adapterRecVHomeworkActAdm.myviewholder>() {
    var dataHolderBackup: ArrayList<dataModelRecVHomeworkActAdm>

    init {
        dataHolderBackup = ArrayList(dataHolder)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.single_row_des_homeworkdefault_fragadm, parent, false)
        return myviewholder(view)
    }

    override fun onBindViewHolder(holder: myviewholder, position: Int) {
        holder.hwTitleHomeworkDefaultFragAdm.text =
            dataHolder[position].hwTitleHomeworkDefaultFragAdm
    }

    override fun getItemCount(): Int {
        return dataHolder.size
    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var hwTitleHomeworkDefaultFragAdm: TextView

        init {
            hwTitleHomeworkDefaultFragAdm =
                itemView.findViewById(R.id.homeTitleTvSingleRdHomeworkDefaultFragAdm)
        }
    }
}
